float64 voltage
r_current_optional current_optional
